### DISCUSSION TITLE
SwiftBar default behavior on clicking menu bar

### DISCUSSION THREAD

**Discussion by edjusted** [swiftbar/SwiftBar#101] [Q&A]
I'm using SwiftBar on Big Sur and I was using BitBar on Mojave. There are two plugins I'm using to control the Music/iTunes app: music-controls-playpause.10s.sh and musicRating.10s.sh

In BitBar/Mojave, when I click on the play/pause button in the menu bar, iTunes plays/pauses. Then I can right-click to get to the BitBar submenu.
And when I click on the # of stars on the musicRating menu bar, it'll set the currently playing song to that many stars. And again I can right-click to get to the submenu.

In SwiftBar/Big Sur, however, a single left click on either plugins causes no actions, but instead opens the submenus.

For the rating menu bar, I can *then* select a star rating on the submenu.
But for the play/pause menu bar, the submenu just shows me SwiftBar. I can't actually play/pause at all.

Is this expected behavior (i.e. left click in SwiftBar is like right-click/option-click in BitBar)? Or am I doing something wrong?

**Comment by melonamin** [comment]
> Is this expected behavior (i.e. left click in SwiftBar is like right-click/option-click in BitBar)? Or am I doing something wrong?

No, you are not missing anything. Currently SwiftBar has two click actions:
* left click opens menu
* left option click opens menu and reveals [hidden menu items](https://github.com/swiftbar/SwiftBar#hiding-default-items)

I wasn't aware of this feature in BitBar, it is an interesting use case, let me look into it

**Reply by bilal-fazlani** [reply]
How about **when there is only one item**,
 left click performs the action (href or bash) and right click reveals the menu with swiftbar details?

The other scenario where top level item has an href and it has an associated drop-down as well, may be action can be ignored..

Just some thoughts.

**Reply by melonamin** [reply]
I think it should be consistent, i.e. if there is an action assigned to the menu bar item it always runs on left click instead of opening the menu.

**Reply by bilal-fazlani** [reply]
Makes sense
